dashboard/app/models/course.rb

```diff
@@ -189,21 +205,30 @@ def self.valid_courses
     end
   end
 
+  # Get the set of valid courses for the dropdown in our sections table, using
+  # any alternate scripts based on any experiments the user belongs to.
+  def self.valid_courses_for_user(user)
```

Comment:
I wonder if we should call this something like `experiment_enabled_courses_for_user(user)` to make it more clear that this should only be called in the case where we care about experiments.